### PR TITLE
Allow TestStatus file to be copied into baselines

### LINKED ISSUE #2760: Copy TestStatus file to the baseline directory
I often find myself wanting to know if a given test passed or failed (entirely or some specific phase) in a previous run of the test suite, for which we have baselines. For example, this would be a quick way of seeing if a test failed the last time around, or it can help indicate if / why there might be problems with the baselines for a given test.

So it would be helpful if the TestStatus file were copied to the baseline directory when running a test suite with the `--generate` flag.

There are a few things I'm not sure of in terms of how this should be implemented:

1. If we copy the TestStatus file in the generate phase (which seems like a logical time to do it), it would be missing some information, since generate comes before some other phases. Would it be acceptable to move the generate phase to be the very last phase, or would that have some negative repercussions?

2. Alternatively, we could copy the TestStatus file as a separate step (probably not even a named phase) at the very end of system_tests_common:run. Would that be reasonable?

3. Is there a way to ensure that this copying of the TestStatus file happens even if there is a failure in the run or some other phase? e.g., would putting this copying in a `finally` block at the end of system_tests_common:run ensure this?

I'm happy to make and test this change myself, but I'd like to get input from @jgfouca , @jedwards4b or others if you have any immediate thoughts. My inclination would be to have a `finally` block at the end of system_tests_common:run that does this copying, so I'd probably try that approach first unless you have other feelings.

### PR DESCRIPTION
**Pull request by jgfouca**
Test suite: scripts_regression_tests
Test baseline: 
Test namelist changes: 
Test status: bit for bit

Fixes #2760 

User interface changes?: N

Update gh-pages html (Y/N)?: N

Code review: @jedwards4b @billsacks

### CODE REVIEW

**Comment by jgfouca** [comment]
@billsacks , I don't have access to a cesm machine, so you might want to try this out.

File: scripts/lib/CIME/test_scheduler.py [hunk 1]
```diff
@@ -921,6 +921,15 @@ def run_tests(self, wait=False,
 
         expect(threading.active_count() == 1, "Leftover threads?")
 
+        # Copy TestStatus files to baselines for tests that have already failed.
```
Comment (billsacks) [inline scripts/lib/CIME/test_scheduler.py]: For my own edification: When does this code get invoked in the whole test process?

File: scripts/lib/CIME/hist_utils.py [hunk 2]
```diff
@@ -387,6 +387,20 @@ def get_extension(model, filepath):
 
     return result
 
+def generate_teststatus(testdir, baseline_dir):
```
Comment (billsacks) [inline scripts/lib/CIME/hist_utils.py]: This function seems kind of out of place in hist_utils, but I'm willing to let it go if you want it here.

**Comment by jgfouca** [comment]
@billsacks , I guess I was making assumptions about the workflow here that were not accurate. I assumed you wanted this as part of a procedural run, not individual tests.

I would expect build failures to appear unless you did the build by hand with case.build.

File: scripts/lib/CIME/test_scheduler.py (hunk 1 above)
Comment (jgfouca) [reply scripts/lib/CIME/test_scheduler.py]: At the end of create_test. So, on a batch machine, once all the tests are built and submitted.

**Pull request merged**